src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl.xaml.cs

```diff
@@ -0,0 +1,315 @@
+            {
+                if (symbol.StartLine >= 0 && symbol.StartLine < Snapshot.LineCount)
+                {
+                    var position = Snapshot.GetLineFromLineNumber(symbol.StartLine).Start.Position;
+                    if (position >= 0 && position <= Snapshot.Length)
+                    {
+                        TextView.Caret.PositionChanged -= FollowCursor;
+                        var point = new SnapshotPoint(Snapshot, position);
+                        var snapshotSpan = new SnapshotSpan(point, point);
+                        TextView.SetSelection(snapshotSpan);
+                        TextView.ViewScroller.EnsureSpanVisible(snapshotSpan);
+                        TextView.Caret.PositionChanged += FollowCursor;
+                    }
+                }
+            }
+        }
+
+        private void FollowCursor(object sender, EventArgs e)
+        {
+            if (Snapshot is not null && TextView is not null && SymbolsTreeItemsSource is not null)
+            {
+                var caretPoint = TextView.GetCaretPoint(Snapshot.TextBuffer);
+                if (caretPoint.HasValue)
+                {
+                    caretPoint.Value.GetLineAndCharacter(out var lineNumber, out var characterIndex);
+                    SymbolsTreeItemsSource.ForEach(node => SelectNode(node, lineNumber, characterIndex));
+                }
+            }
+        }
+
+        private void SelectNode(DocumentSymbolViewModel documentSymbol, int lineNumber, int characterIndex)
+        {
+            var selectedNodeIndex = -1;
+            foreach (var child in documentSymbol.Children)
+            {
+                if (child.StartLine <= lineNumber && child.EndLine >= lineNumber)
+                {
+                    if (child.StartLine == child.EndLine)
+                    {
+                        if (child.StartChar <= characterIndex && child.EndChar >= characterIndex)
+                        {
+                            selectedNodeIndex = documentSymbol.Children.IndexOf(child);
+                        }
+                    }
+                    else
+                    {
+                        selectedNodeIndex = documentSymbol.Children.IndexOf(child);
+                    }
+                }
+            }
+
+            if (selectedNodeIndex != -1)
+            {
+                SelectNode(documentSymbol.Children[selectedNodeIndex], lineNumber, characterIndex);
+            }
+            else if (documentSymbol.StartLine <= lineNumber && documentSymbol.EndLine >= lineNumber)
+            {
+                documentSymbol.IsSelected = true;
+            }
+        }
+
+        internal const int OLECMDERR_E_NOTSUPPORTED = unchecked((int)0x80040100);
+
+        public int QueryStatus(ref Guid pguidCmdGroup, uint cCmds, OLECMD[] prgCmds, IntPtr pCmdText)
+        {
+            // we don't support any commands like rename/undo in this view yet
+            return OLECMDERR_E_NOTSUPPORTED;
```

Comment:
Have we confirmed we don't need to route commands we don't understand? The concern here would be if something like build or F5 stops working if we're focused inside the outline control.

Reply:
Updated so that we no longer route commands.